### ISSUE TITLE
feat: Allow ”as” ES6 keyword for Stencil’s decorators

### ISSUE THREAD

**Issue by splitinfinities** [ionic-team/stencil#3137]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already include this feature request, without success.

### Describe the Feature Request

The `@Event` and `@Element` decorators can be imported and can interfere with global types or be mistakenly used within the components.d.ts file. In those circumstances, it would be helpful for component authors to be able to alias the import like with the `as` keyword of es6 modules. 

```ts
import { h, Event as StencilEvent, Element as StencilElement } from “@stencil/core”;

//etc
```

### Describe the Use Case

This feature allows component authors to use an aliased decorator that would be referred to within the components.d.ts file so that collisions can be reduced when interoping with global browser types. 

### Describe Preferred Solution

Open to perspective here, I’m not an expert on this. 

### Describe Alternatives

Maybe we could rename all the decorators? Or have alternate names that could be imported? 

Open to perspective here, I’m not an expert on this.

**Comment by rwaskiewicz** [comment]
Potentially a part of the same problem/implementation space, but not the same problem: https://github.com/ionic-team/stencil/issues/3037

**Comment by rwaskiewicz** [comment]
Related: https://github.com/ionic-team/stencil-eslint/issues/28, although not the same issue

**Comment by rwaskiewicz** [comment]
This issue has been labeled to be added to our internal backlog to be discussed as a team.

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a part of today's [Stencil 4.9.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.9.0).
